### PR TITLE
fix: fix import tsconfig.json

### PR DESCRIPTION
**Pull request by AliceLanniste**
fix import error:

### CODE REVIEW

**Comment by boris-w** [comment]
Why do you want to remove the `composite`? And changing the `rootDir` and `include` in tsconfig.json is supposed to be for development mode, not for build.

**Comment by AliceLanniste** [comment]
I loaded source code with vs code,there is a import error that different rooDir

**Comment by boris-w** [comment]
> I loaded source code with vs code,there is a import error that different rooDir

Right, this error exists, but `composite` shouldn't have anything to do with it, so it doesn't need to be removed, so no need to remove it. And these errors are only encountered in development mode, so make sure that the build artifacts doesn't change after adding these configurations.

**Comment by AliceLanniste** [comment]
so how to fix this error without affecting build artifacts,or just leave alone?

**Comment by boris-w** [comment]
> so how to fix this error without affecting build artifacts,or just leave alone?

1. No need to delete `composite`
2. When you change tsconfig.json for the development mode, then you have to use tsconfig.build.json when you build, and then tsconfig.build.json to restore the changed `rootDir` and `include`.

Example
```
// tsconfig.json
{
  "compilerOptions": {
    ....
    "rootDir": "../"
  },
  "include": ["./src", "../icons/src"]
}

// tsconfig.build.json
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    ....
    "rootDir": "./src"
  },
  "include": ["./src"]
}
```

**Comment by AliceLanniste** [comment]
> > so how to fix this error without affecting build artifacts,or just leave alone?
> 
> 1. No need to delete `composite`
> 2. When you change tsconfig.json for the development mode, then you have to use tsconfig.build.json when you build, and then tsconfig.build.json to restore the changed `rootDir` and `include`.
> 
> Example
> 
> ```
> // tsconfig.json
> {
>   "compilerOptions": {
>     ....
>     "rootDir": "../"
>   },
>   "include": ["./src", "../icons/src"]
> }
> 
> // tsconfig.build.json
> {
>   "extends": "../../tsconfig.json",
>   "compilerOptions": {
>     ....
>     "rootDir": "./src"
>   },
>   "include": ["./src"]
> }
> ```

thanks a lot . I will fix it

File: packages/ui-lib/tsconfig.json [hunk 1]
```diff
@@ -8,15 +8,14 @@
...
-    "composite": true,
     "paths": {
-      "@teable/icons": ["../../icons/src"],
-    },
+      "@teable/icons": ["../../icons/src"]
+    }
   },
   "exclude": ["**/node_modules", "**/.*/", "dist", "build"],
-  "include": ["./src"],
+  "include": ["./src", "../../icons/src"]
```
Comment (boris-w) [inline packages/ui-lib/tsconfig.json]: This should only be used one level up in the directory.

File: packages/ui-lib/tsconfig.build.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
     "jsx": "react-jsx",
     "jsxFactory": "",
     "jsxFragmentFactory": "",
+    "composite": true,
```
Comment (boris-w) [inline packages/ui-lib/tsconfig.build.json]: There’s no need to add here.

**Comment by AliceLanniste** [comment]
@boris-w  thank you.  I'm very interested in this project. If I want to continue to contribute, where can I start?

**Comment by boris-w** [comment]
> @boris-w thank you. I'm very interested in this project. If I want to continue to contribute, where can I start?

Glad to hear it, I'll add some "Good first issue" when I have some free time.

**Comment by tea-artist** [comment]
This will be fixed in #784

**Pull request closed without merging**